<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Pan UKBB ¬∑ EasyLD.jl</title><meta name="title" content="Pan UKBB ¬∑ EasyLD.jl"/><meta property="og:title" content="Pan UKBB ¬∑ EasyLD.jl"/><meta property="twitter:title" content="Pan UKBB ¬∑ EasyLD.jl"/><meta name="description" content="Documentation for EasyLD.jl."/><meta property="og:description" content="Documentation for EasyLD.jl."/><meta property="twitter:description" content="Documentation for EasyLD.jl."/><script data-outdated-warner src="../../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="../.."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../../assets/documenter.js"></script><script src="../../search_index.js"></script><script src="../../siteinfo.js"></script><script src="../../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href>EasyLD.jl</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>Pan UKBB</a><ul class="internal"><li><a class="tocitem" href="#Downloading"><span>Downloading</span></a></li><li><a class="tocitem" href="#Reading-LD-panel-with-Matrix-interface"><span>Reading LD panel with Matrix interface</span></a></li><li><a class="tocitem" href="#Read-in-a-block-with-get_block"><span>Read in a block with <code>get_block</code></span></a></li><li><a class="tocitem" href="#SNP-information"><span>SNP information</span></a></li></ul></li><li><a class="tocitem" href="../gnomAD/">gnomAD</a></li><li><a class="tocitem" href="../api/">API</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Pan UKBB</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Pan UKBB</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/biona001/EasyLD.jl" title="View the repository on GitHub"><span class="docs-icon fa-brands">ÔÇõ</span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/biona001/EasyLD.jl/blob/main/docs/src/man/pan_UKB.md" title="Edit source on GitHub"><span class="docs-icon fa-solid">ÔÅÑ</span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="Pan-UKB-data"><a class="docs-heading-anchor" href="#Pan-UKB-data">Pan-UKB data</a><a id="Pan-UKB-data-1"></a><a class="docs-heading-anchor-permalink" href="#Pan-UKB-data" title="Permalink"></a></h1><ul><li>Link: https://pan-dev.ukbb.broadinstitute.org/docs/hail-format/index.html#extracting-a-subset-of-ld-matrix</li><li>European samples are the largest, totaling 14.1T of data</li></ul><pre><code class="language-julia hljs">using EasyLD
using CSV
using DataFrames
using Statistics
using LinearAlgebra</code></pre><h2 id="Downloading"><a class="docs-heading-anchor" href="#Downloading">Downloading</a><a id="Downloading-1"></a><a class="docs-heading-anchor-permalink" href="#Downloading" title="Permalink"></a></h2><p>First check how many files there are:</p><pre><code class="language-julia hljs">get_ukb_filenames(&quot;EUR&quot;, join=false)</code></pre><pre><code class="nohighlight hljs">124281-element Vector{String}:
 &quot;part-000000-44-0-0-22f828c8-17c3-7c3c-1fa1-1fc113144aca&quot;
 &quot;part-000001-44-1-0-d35bc353-c9b7-28e7-315b-f1bcd8d9e50b&quot;
 &quot;part-000002-44-2-2-7dc4e8fd-75f7-26d0-4b9c-65628f25cf34&quot;
 &quot;part-000003-44-3-0-74be7ee3-ba85-f9ce-4add-9638cf387f8a&quot;
 &quot;part-000004-44-4-0-b11cbd6a-50e2-4e5e-ae70-932ff3aebd5f&quot;
 &quot;part-000005-44-5-0-104f3537-b7e4-0959-555b-7e65ca2b00ae&quot;
 &quot;part-000006-44-6-0-f99bebaa-7361-9c3e-65a0-2700858c2b28&quot;
 &quot;part-000007-44-7-0-90293a19-7ae6-f6a6-1141-48f62ee6f8b1&quot;
 &quot;part-000008-44-8-0-847d7052-f784-0c5f-c4f8-d16df8745a23&quot;
 &quot;part-000009-44-9-0-2b61d351-aeda-4c03-52aa-cb380599219d&quot;
 &quot;part-000010-44-10-0-90b9d808-fa98-2727-ca3e-d63676464185&quot;
 &quot;part-000011-44-11-0-7a15dbc9-b5c7-3500-8258-886bbf569e35&quot;
 &quot;part-000012-44-12-0-73b49602-7c17-3f3b-860c-6aaa8b25f36b&quot;
 ‚ãÆ
 &quot;part-124226-44-124226-0-9948291d-f1c5-775a-4ed3-7d75c9e7d007&quot;
 &quot;part-124227-44-124227-0-43ffd695-b0c7-577e-0d5a-8ea5f3ed84d0&quot;
 &quot;part-124228-44-124228-0-d22a4e30-04f2-82aa-dfd2-4d3ae95e57b4&quot;
 &quot;part-124229-44-124229-0-12730aef-d105-61a3-5342-f20ed3bf3645&quot;
 &quot;part-124230-44-124230-0-984e649b-fc68-5397-3b47-52b76e87c2c6&quot;
 &quot;part-124231-44-124231-0-9fd5803f-f5c7-1c4c-a5fa-77b76dc1f52d&quot;
 &quot;part-124232-44-124232-0-b891bf57-4536-661b-d679-10aac2532957&quot;
 &quot;part-124233-44-124233-0-1c6f8f3f-9337-fc25-53ef-63e76c5a37cf&quot;
 &quot;part-124234-44-124234-0-ad9c21cc-9c7c-0ccb-9335-921cc35ae152&quot;
 &quot;part-124235-44-124235-0-0f31fad5-1ba4-79b3-dff0-cb17d6b694ed&quot;
 &quot;part-124236-44-124236-0-98383aba-a5c1-8093-c27f-001aae4730d3&quot;
 &quot;part-124237-44-124237-1-36053042-3e3b-0495-e362-d7e756ac8f5a&quot;</code></pre><p>Try downloading the first 10 files (~1GB)</p><pre><code class="language-julia hljs">population = &quot;EUR&quot;
outdir = &quot;/Users/biona001/.julia/dev/EasyLD/data&quot;
download_ukb_LD_matrices(population, outdir, start_from=1, num_files=10)</code></pre><pre><code class="nohighlight hljs">[32mProgress: 100%|‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà| Time: 0:00:49[39m</code></pre><ul><li><p>The result would be saved into outdir directory with name <code>UKBB.EUR.ldadj.bm</code> which itself is a directory.</p></li><li><p>For multithreaded downloads, one can use <code>start_from</code> and <code>num_files</code> keywords to control how many matrices to download (not specifying will download all matrices). A progress meter is automatically displayed.</p></li></ul><p>Finally, download variant index hail table</p><pre><code class="language-julia hljs">population = &quot;EUR&quot;
outdir = &quot;/Users/biona001/.julia/dev/EasyLD/data&quot;
download_ukb_variant_index_tables(population, outdir)</code></pre><p>The result will be stored as <code>UKBB.$population.ldadj.variant.ht</code> in <code>outdir</code></p><h2 id="Reading-LD-panel-with-Matrix-interface"><a class="docs-heading-anchor" href="#Reading-LD-panel-with-Matrix-interface">Reading LD panel with Matrix interface</a><a id="Reading-LD-panel-with-Matrix-interface-1"></a><a class="docs-heading-anchor-permalink" href="#Reading-LD-panel-with-Matrix-interface" title="Permalink"></a></h2><p>The first time <code>hail_block_matrix</code> gets called, we do some pre-processing to the variant index files so subsequent calls will be faster. </p><pre><code class="language-julia hljs">bm_file = &quot;/Users/biona001/.julia/dev/EasyLD/data/UKBB.EUR.ldadj.bm&quot;
ht_file = &quot;/Users/biona001/.julia/dev/EasyLD/data/UKBB.EUR.ldadj.variant.ht&quot;
@time bm = hail_block_matrix(bm_file, ht_file); # the &#39;;&#39; avoids displaying a few entries of bm, which takes ~0.1 seconds per entry</code></pre><pre><code class="nohighlight hljs">130.413312 seconds (978.68 M allocations: 23.643 GiB, 50.65% gc time, 4.97% compilation time)</code></pre><p>Check size of matrix</p><pre><code class="language-julia hljs">size(bm)</code></pre><pre><code class="nohighlight hljs">(23960350, 23960350)</code></pre><p>Read first 10000 by 10000 block into memory</p><pre><code class="language-julia hljs">Sigma = bm[1:10000, 1:10000]</code></pre><pre><code class="nohighlight hljs">[Stage 1:=================================================&gt;         (5 + 1) / 6]




10000√ó10000 Matrix{Float64}:
 0.99996  0.00098559  -0.000830543  ‚Ä¶  -0.000520665   0.00422904
 0.0      0.999942    -0.00104453      -0.00494589    0.00050959
 0.0      0.0          0.999963        -0.00129536    0.00216722
 0.0      0.0          0.0             -0.00663755    0.000142521
 0.0      0.0          0.0              0.0104731     0.000936687
 0.0      0.0          0.0          ‚Ä¶  -0.00202249   -0.000497824
 0.0      0.0          0.0             -0.00550819    0.00283421
 0.0      0.0          0.0             -0.00145102   -0.000761935
 0.0      0.0          0.0              0.00287458   -0.0010154
 0.0      0.0          0.0             -0.000450895  -0.00130722
 0.0      0.0          0.0          ‚Ä¶  -0.00101125    0.000451512
 0.0      0.0          0.0              0.00590455   -0.00110266
 0.0      0.0          0.0             -0.00274252   -0.000817142
 ‚ãÆ                                  ‚ã±                
 0.0      0.0          0.0             -0.109458     -0.00866031
 0.0      0.0          0.0              0.0158602     0.0203563
 0.0      0.0          0.0          ‚Ä¶   0.00745596    0.0184919
 0.0      0.0          0.0              0.983906     -0.0122066
 0.0      0.0          0.0              0.0155851     0.0197608
 0.0      0.0          0.0              0.965104     -0.0159355
 0.0      0.0          0.0             -0.129534      0.088277
 0.0      0.0          0.0          ‚Ä¶   0.622303      0.059004
 0.0      0.0          0.0              0.965347     -0.0160028
 0.0      0.0          0.0              0.0925609     0.999286
 0.0      0.0          0.0              0.999972      0.0926374
 0.0      0.0          0.0              0.0           0.999974</code></pre><p>Check if the given block is PSD by computing its eigenvalues</p><pre><code class="language-julia hljs">eigvals(Symmetric(Sigma)) # Symmetric uses upper triangular portion of data</code></pre><pre><code class="nohighlight hljs">10000-element Vector{Float64}:
  -4.496303144059796e-15
  -1.762987426134658e-15
  -1.040030990442209e-15
  -4.2089306816541705e-16
  -3.960324780237165e-16
  -3.9060196040323796e-16
  -2.985734642577166e-16
  -2.8601497287222336e-16
  -2.5553650845256305e-16
  -1.998130772222011e-16
  -1.8449274904356797e-16
  -1.7028015609869147e-16
  -1.3192630645503314e-16
   ‚ãÆ
 116.57001725005479
 124.37732317876345
 125.11477060743002
 135.80425399688727
 142.93300594996916
 149.24000792252738
 161.47927858350278
 181.82233513932786
 228.3300216753156
 235.961635959381
 280.6540947869515
 322.07381500507836</code></pre><h2 id="Read-in-a-block-with-get_block"><a class="docs-heading-anchor" href="#Read-in-a-block-with-get_block">Read in a block with <code>get_block</code></a><a id="Read-in-a-block-with-get_block-1"></a><a class="docs-heading-anchor-permalink" href="#Read-in-a-block-with-get_block" title="Permalink"></a></h2><p>One can also extract a block by specifying the chromosome and starting/ending basepair</p><pre><code class="language-julia hljs">chr = 1
start_pos = 11063
end_pos = 91588
sigma, df = get_block(bm, chr, start_pos, end_pos; min_maf=0.0)
sigma</code></pre><pre><code class="nohighlight hljs">9√ó9 Matrix{Float64}:
 0.99996  0.00098559  -0.000830543  ‚Ä¶  -0.00045888    0.000221516
 0.0      0.999942    -0.00104453      -0.000332668  -0.000439517
 0.0      0.0          0.999963        -0.000551892  -0.000786793
 0.0      0.0          0.0             -0.000503539  -0.000712834
 0.0      0.0          0.0             -0.000294435   0.00575015
 0.0      0.0          0.0          ‚Ä¶  -0.000112519  -0.00015419
 0.0      0.0          0.0             -0.000264108   0.00413586
 0.0      0.0          0.0              0.999978     -0.000225057
 0.0      0.0          0.0              0.0           0.99995</code></pre><pre><code class="language-julia hljs"># SNP information of this block
@show df;</code></pre><pre><code class="nohighlight hljs">df = 9√ó6 DataFrame
 Row ‚îÇ rsid         AF          chr      pos    ref     alt
     ‚îÇ String       Float64     String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ rs561109771  4.7982e-5   1        11063  T       G
   2 ‚îÇ rs562993331  0.00027798  1        13259  G       A
   3 ‚îÇ rs578081284  0.00083096  1        17641  G       A
   4 ‚îÇ rs576081345  0.00065859  1        57222  T       C
   5 ‚îÇ rs570371753  0.00024023  1        58396  T       C
   6 ‚îÇ rs561430336  2.7728e-5   1        63668  G       A
   7 ‚îÇ rs2531267    0.00018542  1        69569  T       C
   8 ‚îÇ rs557418932  8.1599e-5   1        79192  T       G
   9 ‚îÇ rs554639997  0.00015868  1        91588  G       A</code></pre><p>When importing blocks, one can filter for minimum minor allele frequency</p><pre><code class="language-julia hljs"># keep SNPs with MAF &gt; 0.0001
chr = 1
start_pos = 11063
end_pos = 91588
sigma, df = get_block(bm, chr, start_pos, end_pos; min_maf=0.0001)
sigma</code></pre><pre><code class="nohighlight hljs">6√ó6 Matrix{Float64}:
 0.999942  -0.00104453  -0.000939341  ‚Ä¶  -0.000502128  -0.000439517
 0.0        0.999963    -0.00164293      -0.000874333  -0.000786793
 0.0        0.0          0.999972        -0.00079952   -0.000712834
 0.0        0.0          0.0             -0.00044672    0.00575015
 0.0        0.0          0.0              0.999986      0.00413586
 0.0        0.0          0.0          ‚Ä¶   0.0           0.99995</code></pre><pre><code class="language-julia hljs">@show df;</code></pre><pre><code class="nohighlight hljs">df = 6√ó6 DataFrame
 Row ‚îÇ rsid         AF          chr      pos    ref     alt
     ‚îÇ String       Float64     String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ rs562993331  0.00027798  1        13259  G       A
   2 ‚îÇ rs578081284  0.00083096  1        17641  G       A
   3 ‚îÇ rs576081345  0.00065859  1        57222  T       C
   4 ‚îÇ rs570371753  0.00024023  1        58396  T       C
   5 ‚îÇ rs2531267    0.00018542  1        69569  T       C
   6 ‚îÇ rs554639997  0.00015868  1        91588  G       A</code></pre><p>One can also provide a list of SNP positions, and we will only keep SNPs that have those position which also pass the <code>min_maf</code> filter</p><pre><code class="language-julia hljs"># keep SNPs with MAF &gt; 0.0001 and only include a list of SNPs with known positions
chr = 1
start_pos = 11063
end_pos = 91588
snps_to_keep = [13259, 58396, 91588]
sigma, df = get_block(bm, chr, start_pos, end_pos; min_maf=0.0001, snps_to_keep=snps_to_keep)
sigma</code></pre><pre><code class="nohighlight hljs">[Stage 2:&gt;                                                          (0 + 1) / 1]




3√ó3 Matrix{Float64}:
 0.999942  -0.00056507  -0.000439517
 0.0        0.999982     0.00575015
 0.0        0.0          0.99995</code></pre><pre><code class="language-julia hljs">@show df;</code></pre><pre><code class="nohighlight hljs">df = 3√ó6 DataFrame
 Row ‚îÇ rsid         AF          chr      pos    ref     alt
     ‚îÇ String       Float64     String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ rs562993331  0.00027798  1        13259  G       A
   2 ‚îÇ rs570371753  0.00024023  1        58396  T       C
   3 ‚îÇ rs554639997  0.00015868  1        91588  G       A</code></pre><h2 id="SNP-information"><a class="docs-heading-anchor" href="#SNP-information">SNP information</a><a id="SNP-information-1"></a><a class="docs-heading-anchor-permalink" href="#SNP-information" title="Permalink"></a></h2><p>SNP information can be accessed from the <code>bm.info</code> field, where <code>AF</code> corresponds to alternate allele frequency. </p><pre><code class="language-julia hljs">@show bm.info[1:10, :]; # first 10 rows</code></pre><pre><code class="nohighlight hljs">bm.info[1:10, :] = 10√ó6 DataFrame
 Row ‚îÇ rsid         AF          chr      pos     ref     alt
     ‚îÇ String       Float64     String3  Int64   String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ rs561109771  4.7982e-5   1         11063  T       G
   2 ‚îÇ rs562993331  0.00027798  1         13259  G       A
   3 ‚îÇ rs578081284  0.00083096  1         17641  G       A
   4 ‚îÇ rs576081345  0.00065859  1         57222  T       C
   5 ‚îÇ rs570371753  0.00024023  1         58396  T       C
   6 ‚îÇ rs561430336  2.7728e-5   1         63668  G       A
   7 ‚îÇ rs2531267    0.00018542  1         69569  T       C
   8 ‚îÇ rs557418932  8.1599e-5   1         79192  T       G
   9 ‚îÇ rs554639997  0.00015868  1         91588  G       A
  10 ‚îÇ rs575442534  0.00066441  1        533573  G       A</code></pre><p>Alternatively, one can read SNP information by giving the variant index folder</p><pre><code class="language-julia hljs">df = read_variant_index_tables(ht_file)</code></pre></article><nav class="docs-footer"><a class="docs-footer-nextpage" href="../gnomAD/">gnomAD ¬ª</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.5.0 on <span class="colophon-date" title="Thursday 27 June 2024 02:37">Thursday 27 June 2024</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
