<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>API · EasyLD.jl</title><meta name="title" content="API · EasyLD.jl"/><meta property="og:title" content="API · EasyLD.jl"/><meta property="twitter:title" content="API · EasyLD.jl"/><meta name="description" content="Documentation for EasyLD.jl."/><meta property="og:description" content="Documentation for EasyLD.jl."/><meta property="twitter:description" content="Documentation for EasyLD.jl."/><script data-outdated-warner src="../../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="../.."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../../assets/documenter.js"></script><script src="../../search_index.js"></script><script src="../../siteinfo.js"></script><script src="../../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../pan_UKB/">EasyLD.jl</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../pan_UKB/">Pan UKBB</a></li><li><a class="tocitem" href="../gnomAD/">gnomAD</a></li><li class="is-active"><a class="tocitem" href>API</a><ul class="internal"><li><a class="tocitem" href="#Exported-functions"><span>Exported functions</span></a></li><li><a class="tocitem" href="#Internal-helper-functions-and-structs"><span>Internal helper functions and structs</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>API</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>API</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/biona001/EasyLD.jl" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/biona001/EasyLD.jl/blob/main/docs/src/man/api.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="API"><a class="docs-heading-anchor" href="#API">API</a><a id="API-1"></a><a class="docs-heading-anchor-permalink" href="#API" title="Permalink"></a></h1><p>Here is a list of available function calls. A detailed description can be found below. </p><ul><li><a href="#EasyLD.HailBlockMatrix"><code>EasyLD.HailBlockMatrix</code></a></li><li><a href="#EasyLD._extract_alternate_allele_freq"><code>EasyLD._extract_alternate_allele_freq</code></a></li><li><a href="#EasyLD._extract_genome_build"><code>EasyLD._extract_genome_build</code></a></li><li><a href="#EasyLD._extract_ref_alt_alleles"><code>EasyLD._extract_ref_alt_alleles</code></a></li><li><a href="#EasyLD._read_metadata"><code>EasyLD._read_metadata</code></a></li><li><a href="#EasyLD.download_gnomad_LD_matrices"><code>EasyLD.download_gnomad_LD_matrices</code></a></li><li><a href="#EasyLD.download_gnomad_variant_index_tables"><code>EasyLD.download_gnomad_variant_index_tables</code></a></li><li><a href="#EasyLD.download_ukb_LD_matrices"><code>EasyLD.download_ukb_LD_matrices</code></a></li><li><a href="#EasyLD.download_ukb_variant_index_tables"><code>EasyLD.download_ukb_variant_index_tables</code></a></li><li><a href="#EasyLD.get_block"><code>EasyLD.get_block</code></a></li><li><a href="#EasyLD.get_gnomad_filenames"><code>EasyLD.get_gnomad_filenames</code></a></li><li><a href="#EasyLD.get_ukb_filenames"><code>EasyLD.get_ukb_filenames</code></a></li><li><a href="#EasyLD.hail_block_matrix"><code>EasyLD.hail_block_matrix</code></a></li><li><a href="#EasyLD.read_variant_index_tables"><code>EasyLD.read_variant_index_tables</code></a></li></ul><h2 id="Exported-functions"><a class="docs-heading-anchor" href="#Exported-functions">Exported functions</a><a id="Exported-functions-1"></a><a class="docs-heading-anchor-permalink" href="#Exported-functions" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.download_gnomad_LD_matrices" href="#EasyLD.download_gnomad_LD_matrices"><code>EasyLD.download_gnomad_LD_matrices</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_gnomad_LD_matrices(population::String, outdir::String, [start_from], [num_files])</code></pre><p>Downloads the LD matrices from gnomAD panel for specified population and saves result to <code>outdir</code>. See <code>https://gnomad.broadinstitute.org/downloads#v2-linkage-disequilibrium</code>.</p><p><strong>Inputs</strong></p><ul><li><code>population</code>: Options include<ul><li><code>afr</code> for African/African American population</li><li><code>amr</code> for Latino/Admixed American population</li><li><code>asj</code> for Ashkenazi Jewish population</li><li><code>eas</code> for East Asian population</li><li><code>fin</code> for European (Finnish) population</li><li><code>nfe</code> for European (non-Finnish) population</li><li><code>est</code> for Estonian population</li><li><code>nwe</code> for North-western European population</li><li><code>seu</code> for Southern European population</li></ul></li><li><code>outdir</code>: Directory for which files will be saved.</li><li><code>start_from</code>: Int specifying which file to start downloading at</li><li><code>num_files</code>: Int specifying how many files to download</li></ul><p><strong>Multiple download sessions at once</strong></p><p>Since there are lots of files to download, one can specify the number of files to download by modifying the <code>num_files</code> files option. This function will download the first <code>num_files</code> file counting from the <code>start_from</code> element  of the output of <code>get_gnomad_filenames()</code>. For example, if  <code>myfiles = get_gnomad_filenames(&quot;afr&quot;)</code>, then the function will download files  from <code>myfiles[start_from]</code> through <code>myfiles[start_from+num_files-1]</code></p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/download.jl#L1-L29">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.download_ukb_LD_matrices" href="#EasyLD.download_ukb_LD_matrices"><code>EasyLD.download_ukb_LD_matrices</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_ukb_LD_matrices(population::String, outdir::String, [start_from], [num_files])</code></pre><p>Downloads the LD matrices from pan-UK-biobank for specified population and saves result to <code>outdir</code>. See  <code>https://pan-dev.ukbb.broadinstitute.org/docs/hail-format/index.html</code>.</p><p><strong>Inputs</strong></p><ul><li><code>population</code>: Several options are available. These codes refer only to ancestry    groupings used in GWAS, not necessarily other demographic or self-reported data.<ul><li><code>EUR</code> for European ancestry</li><li><code>CSA</code> for Central/South Asian ancestry</li><li><code>AFR</code> for African ancestry</li><li><code>EAS</code> for East Asian ancestry</li><li><code>MID</code> for Middle Eastern ancestry</li><li><code>AMR</code> for Admixed American ancestry</li></ul></li><li><code>outdir</code>: Directory for which files will be saved.</li><li><code>start_from</code>: Int specifying which file to start downloading at</li><li><code>num_files</code>: Int specifying how many files to download</li></ul><p><strong>Multiple download sessions at once</strong></p><p>Since there are lots of files to download, one can specify the number of files to download by modifying the <code>num_files</code> files option. This function will download the first <code>num_files</code> file counting from the <code>start_from</code> element  of the output of <code>get_ukb_filenames()</code>. For example, if  <code>myfiles = get_ukb_filenames(&quot;afr&quot;)</code>, then the function will download files  from <code>myfiles[start_from]</code> through <code>myfiles[start_from+num_files-1]</code></p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/download.jl#L61-L88">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.download_gnomad_variant_index_tables" href="#EasyLD.download_gnomad_variant_index_tables"><code>EasyLD.download_gnomad_variant_index_tables</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_gnomad_variant_index_tables(population::String, outdir::String)</code></pre><p>Downloads variant index hail tables from gnomAD panel for specified population  and saves result to <code>outdir</code>. <code>population</code> accepts the same population labels as  <code>download_gnomad_LD_matrices</code>. These files are typically at most a few GB.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/download.jl#L163-L169">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.download_ukb_variant_index_tables" href="#EasyLD.download_ukb_variant_index_tables"><code>EasyLD.download_ukb_variant_index_tables</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_ukb_variant_index_tables(population::String, outdir::String)</code></pre><p>Downloads variant index hail tables from pan-UK-biobank for specified population  and saves result to <code>outdir</code>. <code>population</code> accepts the same population labels as  <code>download_ukb_LD_matrices</code>. These files are typically at most a few GB.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/download.jl#L146-L152">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.get_gnomad_filenames" href="#EasyLD.get_gnomad_filenames"><code>EasyLD.get_gnomad_filenames</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_gnomad_filenames(population::String)</code></pre><p>Returns a list of file names that would be downloaded for the specified population. This function will not download anything. If <code>join=true</code>, the full path will be returned</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/download.jl#L120-L125">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.get_ukb_filenames" href="#EasyLD.get_ukb_filenames"><code>EasyLD.get_ukb_filenames</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_ukb_filenames(population::String)</code></pre><p>Returns a list of file names that would be downloaded for the specified population. This function will not download anything. If <code>join=true</code>, the full path will be returned</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/download.jl#L133-L138">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.hail_block_matrix" href="#EasyLD.hail_block_matrix"><code>EasyLD.hail_block_matrix</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">hail_block_matrix(bm_files::String, ht_files::String)</code></pre><p>Creates a <code>HailBlockMatrix</code> which allows reading chunks of LD matrix data into  memory. It is a subtype of <code>AbstractMatrix</code>, so operations like indexing and  <code>size()</code> works, but only accessing its elements in contiguous chunks is &quot;fast&quot;.  I may support more functionalities depending on interest. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">using EasyLD
datadir = &quot;/Users/biona001/.julia/dev/EasyLD/data&quot;
bm_file = joinpath(datadir, &quot;UKBB.EUR.ldadj.bm&quot;)
ht_file = joinpath(datadir, &quot;UKBB.EUR.ldadj.variant.ht&quot;)
bm = hail_block_matrix(bm_file, ht_file); # need a &#39;;&#39; to avoid displaying a few entries of bm, which takes ~0.1 seconds per entry

# get matrix dimension
size(bm) # returns (23960350, 23960350)

# read a single entry
bm[1, 1] # returns 0.999959841549239

# read first 10k by 10k block into memory (takes roughly 7 seconds)
bm[1:10000, 1:10000]

# arbitrary slicing works but is very slow
bm[1:3, 1:2:100] # ~22 seconds

# read a specific chromosome region
chr = 1
start_pos = 11063
end_pos = 91588
sigma = get_block(bm, chr, start_pos, end_pos)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L135-L170">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.read_variant_index_tables" href="#EasyLD.read_variant_index_tables"><code>EasyLD.read_variant_index_tables</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">read_variant_index_tables(ht_file::String; [alt_allele_header], [reprocess])</code></pre><p>Read variant index hail tables into a <code>DataFrame</code>. The first time this function gets called, we will read the original <code>.ht</code> files into memory and write the result to a tab-separated value <code>.tsv</code> file into the same directory as <code>ht_file</code>,  and we will also create a comma separated file ending in <code>.csv</code> which pre-process the <code>.tsv</code> file for easier reading. If <code>reprocess=true</code>, we will re-run the preprocessing step.</p><p><strong>Note</strong></p><ul><li><code>rsIDs</code>: Some panels (e.g. Pan-UKB) also contain rsID, but others (e.g. gnomAD) does not,   so rsID may not be available. </li><li><code>AF</code>: Alternate Allele frequency is sometimes available in the <code>.tsv</code> file in   its own column (e.g. Pan-UKB), but other times it&#39;s hidden inside a column    of the <code>.tsv</code> file (e.g. in gnomAD it is under the header <code>pop_freq</code> which   includes <code>AF</code> but also other information). In the later case, we will assume   the column in <code>.tsv</code> has header name <code>alt_allele_header</code> and is JSON   formatted, and further that alt-allele freq is available as <code>AF</code> key. See   <a href="#EasyLD._extract_alternate_allele_freq"><code>_extract_alternate_allele_freq</code></a></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L193-L213">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.get_block" href="#EasyLD.get_block"><code>EasyLD.get_block</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_block(bm::HailBlockMatrix, chr, start_bp, end_bp; [min_maf], 
    [snps_to_keep], [min_eigval], [enforce_psd], [rk])</code></pre><p>Reads in a block of LD matrix from chromosome <code>chr</code> between basepairs  <code>start_bp</code> and <code>end_bp</code>. The inputs <code>chr</code>, <code>start_bp</code>, <code>end_bp</code> can be Int or String. The result will always be a matrix even if <code>start_bp == end_bp</code>. If  <code>start_bp</code> or <code>end_bp</code> is not in the LD matrix, we will return the smallest region that does NOT include them. For example, if  <code>(start_bp, end_bp) = (555, 777)</code> and SNP positions in the LD panel are  <code>positions = [..., 400,  500,  600,  700,  800,  900, ...]</code> with <code>snp_names = [..., snp4, snp5, snp6, snp7, snp8, snp9, ..]</code>,  then we will return the LD matrix for <code>[snp6, snp7]</code></p><p><strong>Inputs</strong></p><ul><li><code>bm</code>: A <code>HailBlockMatrix</code></li><li><code>chr</code>: Chromosome number, can be an Int or String (e.g. <code>1</code> or <code>&quot;1&quot;</code>)</li><li><code>start_bp</code>: Starting basepair, can be an Int or String</li><li><code>end_bp</code>: Ending basepair, can be an Int or String</li></ul><p><strong>Optional inputs</strong></p><ul><li><code>min_maf</code>: Minimum minor allele frequency. Only variants with alternate allele   frequency between <code>[min_maf, 1-min_maf]</code> is kept. Default <code>min_maf=0.01</code></li><li><code>snps_to_keep</code>: Vector of SNP positions to import. If both <code>snps_to_keep</code> and   <code>min_maf</code> are specified, only SNPs whose position is listed in <code>snps_to_keep</code>   whose minor allele frequency exceeds <code>min_maf</code> will be kept. </li><li><code>min_eigval</code>: Smallest eigenvalue allowed in Σ. All eigenvalues smaller than    <code>min_eigval</code> will be set to <code>min_eigval</code> (default <code>1e-5</code>). This option is only   used if <code>enforce_psd=true</code>.</li><li><code>enforce_psd</code>: LD data stored in Pan-UKB or gnomAD LD panels only includes the   upper triangular portion. If <code>enforce_psd</code> is true, we will copy the upper    triangular portion to the lower triangular portion, force eigenvalues to be   above <code>min_eigval</code>, and then scale the covariance matrix into a correlation   matrix (default <code>true</code>)</li><li><code>rk</code>: An <code>Int</code>. If specified and <code>enforce_psd=true</code>, we will keep only the top   <code>rk</code> eigenvalues of imported <code>Σ</code> before truncating the rest to <code>min_eigval</code>.   This experiments with the &quot;truncated-SVD&quot; approach of    <code>https://www.sciencedirect.com/science/article/pii/S0002929717303919#bib19</code>   (default <code>Inf</code>)</li></ul><p><strong>Note</strong></p><p>Make sure <code>start_bp</code> and <code>end_bp</code> is from the same human genome build as the  LD matrices. One can verify build information via <code>bm.build</code>. Both Pan-UKBB and  gnomAD uses hg19 (i.e. GRCh37).</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L43-L89">source</a></section></article><h2 id="Internal-helper-functions-and-structs"><a class="docs-heading-anchor" href="#Internal-helper-functions-and-structs">Internal helper functions and structs</a><a id="Internal-helper-functions-and-structs-1"></a><a class="docs-heading-anchor-permalink" href="#Internal-helper-functions-and-structs" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD.HailBlockMatrix" href="#EasyLD.HailBlockMatrix"><code>EasyLD.HailBlockMatrix</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Currently a <code>HailBlockMatrix</code> is just a python wrapper, where data reading is  achieved within python by the Hail software, then the result is passed into Julia via PyCall.jl. The <code>bm</code> field behaves as a <code>BlockMatrix</code> from hail. The <code>info</code> fields stores the <code>chr/pos/ref/alt/AF</code> information, where <code>pos</code> uses coordinates in genome build <code>build</code>.</p><p>The Hail Block Matrix format is described here https://hail.is/docs/0.2/linalg/hail.linalg.BlockMatrix.html#blockmatrix</p><p>A more specific file description is provided here https://discuss.hail.is/t/blockmatrix-specification/3118</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L1-L13">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD._extract_alternate_allele_freq" href="#EasyLD._extract_alternate_allele_freq"><code>EasyLD._extract_alternate_allele_freq</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">_extract_alternate_allele_freq(df::DataFrame; header_name = &quot;pop_freq&quot;)</code></pre><p>Internal helper function that extracts alternate allele frequency from the  <code>ht_file</code> assuming this information is present in the variant index files  as JSON format with header <code>header_name</code>. Only <code>gnomAD</code> panel will actually reach here since Pan-UKB has <code>AF</code> as a separate field in its variant index files</p><p>For example, in gnomAD, we have</p><pre><code class="language-julia hljs">julia&gt; tsv_file = &quot;gnomad.genomes.r2.1.1.nfe.common.adj.ld.variant_indices.ht/variant.ht.tsv&quot;
julia&gt; df = CSV.read(tsv_file, DataFrame)
julia&gt; names(df)
4-element Vector{String}:
 &quot;locus&quot;
 &quot;alleles&quot;
 &quot;pop_freq&quot;
 &quot;idx&quot;
julia&gt; df[1, &quot;pop_freq&quot;] # first entry with header `pop_freq` that contains AF
&quot;{&quot;AC&quot;:861,&quot;AF&quot;:0.40922053231939165,&quot;AN&quot;:2104,&quot;homozygote_count&quot;:166}&quot;</code></pre><p>Then the AF field for all variants will be extracted</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L255-L277">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD._extract_ref_alt_alleles" href="#EasyLD._extract_ref_alt_alleles"><code>EasyLD._extract_ref_alt_alleles</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">_extract_ref_alt_alleles(alleles::Vector{String})</code></pre><p>Internal helper function to extract the ref/alt alleles from hail tables. Each  element of <code>alleles</code> should be a <code>string</code> that has 2 alleles wrapped in quotations, as in the case for Pan-UKB and gnomAD. The first is assumed to be reference and  the second is alt. As an example, <code>[&quot;CCTAA&quot;,&quot;C&quot;]</code> with give <code>ref = &quot;CCTAA&quot;</code> and  <code>alt = &quot;C&quot;</code>. </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L286-L294">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD._extract_genome_build" href="#EasyLD._extract_genome_build"><code>EasyLD._extract_genome_build</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">_extract_genome_build(ht_file::String)</code></pre><p>Internal helper function to extracts the human genome build information from  the variant index folder. We assume this information is stored in a file called  <code>metadata.json</code> or <code>metadata.json.gz</code> in the following format <code>Locus(XXX)</code> where <code>XXX</code> is the reference build. If this format is not detected, the genome build information will be &quot;Unknown&quot;</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L313-L321">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="EasyLD._read_metadata" href="#EasyLD._read_metadata"><code>EasyLD._read_metadata</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">_read_metadata(metafile::String)</code></pre><p>Internal helpder function that reads the metadata file and returns a dictionary. <code>metafile</code> should end with <code>.json</code> or <code>.json.gz</code></p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/biona001/EasyLD.jl/blob/0e4d6957016376d2c7c6d251f044ecdb20fa8535/src/hailBM.jl#L334-L339">source</a></section></article></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../gnomAD/">« gnomAD</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.5.0 on <span class="colophon-date" title="Friday 28 June 2024 02:38">Friday 28 June 2024</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
