<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>gnomAD ¬∑ EasyLD.jl</title><meta name="title" content="gnomAD ¬∑ EasyLD.jl"/><meta property="og:title" content="gnomAD ¬∑ EasyLD.jl"/><meta property="twitter:title" content="gnomAD ¬∑ EasyLD.jl"/><meta name="description" content="Documentation for EasyLD.jl."/><meta property="og:description" content="Documentation for EasyLD.jl."/><meta property="twitter:description" content="Documentation for EasyLD.jl."/><script data-outdated-warner src="../../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="../.."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../../assets/documenter.js"></script><script src="../../search_index.js"></script><script src="../../siteinfo.js"></script><script src="../../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../pan_UKB/">EasyLD.jl</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../pan_UKB/">Pan UKBB</a></li><li class="is-active"><a class="tocitem" href>gnomAD</a><ul class="internal"><li><a class="tocitem" href="#Downloading"><span>Downloading</span></a></li><li><a class="tocitem" href="#Reading-LD-panel-with-Matrix-interface"><span>Reading LD panel with Matrix interface</span></a></li><li><a class="tocitem" href="#Read-in-a-block-with-get_block"><span>Read in a block with <code>get_block</code></span></a></li><li><a class="tocitem" href="#SNP-information"><span>SNP information</span></a></li></ul></li><li><a class="tocitem" href="../api/">API</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>gnomAD</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>gnomAD</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/biona001/EasyLD.jl" title="View the repository on GitHub"><span class="docs-icon fa-brands">ÔÇõ</span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/biona001/EasyLD.jl/blob/main/docs/src/man/gnomAD.md" title="Edit source on GitHub"><span class="docs-icon fa-solid">ÔÅÑ</span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="gnomAD-data"><a class="docs-heading-anchor" href="#gnomAD-data">gnomAD data</a><a id="gnomAD-data-1"></a><a class="docs-heading-anchor-permalink" href="#gnomAD-data" title="Permalink"></a></h1><ul><li>Link: https://gnomad.broadinstitute.org/downloads#v2-linkage-disequilibrium</li><li><code>nfe</code> population totals ~555GB</li></ul><pre><code class="language-julia hljs">using Revise
using EasyLD
using CSV
using DataFrames
using Statistics
using LinearAlgebra</code></pre><h2 id="Downloading"><a class="docs-heading-anchor" href="#Downloading">Downloading</a><a id="Downloading-1"></a><a class="docs-heading-anchor-permalink" href="#Downloading" title="Permalink"></a></h2><p>First check how many files there are:</p><pre><code class="language-julia hljs">get_gnomad_filenames(&quot;nfe&quot;, join=false)</code></pre><pre><code class="nohighlight hljs">10023-element Vector{String}:
 &quot;part-00000-14-0-0-7e282c91-9060-db76-1d4a-a02877b62910&quot;
 &quot;part-00001-14-1-0-2bb43364-b9d9-7048-c4e0-74c951357d81&quot;
 &quot;part-00002-14-2-0-998bf016-ab2f-aaf9-ce80-7888e4ee7524&quot;
 &quot;part-00003-14-3-0-5ad64a0b-ecb1-cf91-69ab-e3ee2fcdef51&quot;
 &quot;part-00004-14-4-0-7a03381b-dc38-5f21-f8dd-ad869dd1f340&quot;
 &quot;part-00005-14-5-0-d78f2159-d71c-adc2-47eb-5af6a6c34015&quot;
 &quot;part-00006-14-6-0-dc8b5638-9246-15b8-579e-d14ff6a69646&quot;
 &quot;part-00007-14-7-0-f41e5872-2ed6-36d2-e625-bb6d6261c6bc&quot;
 &quot;part-00008-14-8-0-27a9b5b4-b329-d7a8-9d86-f3184776cb09&quot;
 &quot;part-00009-14-9-0-68655a89-424f-0cb7-b35c-583b52c859e3&quot;
 &quot;part-00010-14-10-0-a60938ee-75ec-91c9-91f3-fbcffa73a20e&quot;
 &quot;part-00011-14-11-0-3be840ef-ae93-3775-47fe-fc33eedb9911&quot;
 &quot;part-00012-14-12-0-8cbce5b3-023a-b252-52c2-cc9c717d0ec5&quot;
 ‚ãÆ
 &quot;part-10011-14-10011-0-1ed25fc8-64a3-7084-60be-717f4c447fac&quot;
 &quot;part-10012-14-10012-0-342cfc1b-d185-4c2d-ad6e-d9595c5b9071&quot;
 &quot;part-10013-14-10013-0-004a90a8-ec7c-0664-d592-47686c3df576&quot;
 &quot;part-10014-14-10014-0-fdad432a-59ea-ef55-aa6c-dfebf128272c&quot;
 &quot;part-10015-14-10015-0-5216889b-97ba-8e71-8d77-27c2e41a7a38&quot;
 &quot;part-10016-14-10016-0-955e1d70-ef0f-5dc7-cf3f-55c2d7626c53&quot;
 &quot;part-10017-14-10017-0-14ea97f3-00af-93ba-ffa7-f6a0ff9c6541&quot;
 &quot;part-10018-14-10018-0-2cc71e56-18a8-de9e-85c8-1513ec63cceb&quot;
 &quot;part-10019-14-10019-0-16de6180-8034-d811-4875-a91b2da0dad8&quot;
 &quot;part-10020-14-10020-0-1a65d50c-ffe5-2f10-c510-60e0cfa5b186&quot;
 &quot;part-10021-14-10021-0-fc83fd8d-e05e-6955-fc58-bcf4296985ba&quot;
 &quot;part-10022-14-10022-0-216231b4-e554-eac9-d165-ffa8a90fef05&quot;</code></pre><pre><code class="language-julia hljs">population = &quot;nfe&quot;
outdir = &quot;/Users/biona001/.julia/dev/EasyLD/data&quot;
download_gnomad_LD_matrices(population, outdir, start_from=1, num_files=10)</code></pre><pre><code class="nohighlight hljs">[32mProgress: 100%|‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà| Time: 0:00:29[39m</code></pre><ul><li><p>The result would be saved into outdir directory with name <code>gnomad.genomes.r2.1.1.nfe.common.adj.ld.bm</code> which itself is a directory.</p></li><li><p>For multithreaded downloads, one can use <code>start_from</code> and <code>num_files</code> keywords to control how many matrices to download (not specifying will download all matrices). A progress meter is automatically displayed.</p></li></ul><p>One also needs the variant index files which tells us chr/pos/ref/alt/SNP-name/alt-allele-frequency</p><pre><code class="language-julia hljs">population = &quot;nfe&quot;
outdir = &quot;/Users/biona001/.julia/dev/EasyLD/data&quot;
@time download_gnomad_variant_index_tables(population, outdir)</code></pre><p>The result will be stored as <code>gnomad.genomes.r2.1.1.$population.common.adj.ld.variant_indices.ht</code> in <code>outdir</code></p><h2 id="Reading-LD-panel-with-Matrix-interface"><a class="docs-heading-anchor" href="#Reading-LD-panel-with-Matrix-interface">Reading LD panel with Matrix interface</a><a id="Reading-LD-panel-with-Matrix-interface-1"></a><a class="docs-heading-anchor-permalink" href="#Reading-LD-panel-with-Matrix-interface" title="Permalink"></a></h2><p>The first time <code>hail_block_matrix</code> gets called, we do some pre-processing to the variant index files so subsequent calls will be faster. </p><pre><code class="language-julia hljs">bm_file = &quot;/Users/biona001/.julia/dev/EasyLD/data/gnomad.genomes.r2.1.1.nfe.common.adj.ld.bm&quot;
ht_file = &quot;/Users/biona001/.julia/dev/EasyLD/data/gnomad.genomes.r2.1.1.nfe.common.adj.ld.variant_indices.ht&quot;
@time bm = hail_block_matrix(bm_file, ht_file); # the &#39;;&#39; avoids displaying a few entries of bm, which takes ~0.1 seconds per entry</code></pre><pre><code class="nohighlight hljs"> 33.650270 seconds (477.52 M allocations: 10.556 GiB, 12.67% gc time, 18.13% compilation time)</code></pre><p>Check size of matrix</p><pre><code class="language-julia hljs">size(bm)</code></pre><pre><code class="nohighlight hljs">(14207204, 14207204)</code></pre><p>Read first 10000 by 10000 block into memory</p><pre><code class="language-julia hljs">Sigma = bm[1:10000, 1:10000]</code></pre><pre><code class="nohighlight hljs">[Stage 0:=================================================&gt;         (5 + 1) / 6]




10000√ó10000 Matrix{Float64}:
 1.0  0.0326361  -0.0181891   ‚Ä¶   0.0         0.0         0.0
 0.0  1.0        -0.00041394      0.0         0.0         0.0
 0.0  0.0         1.0             0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0         ‚Ä¶   0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0         ‚Ä¶   0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 0.0  0.0         0.0             0.0         0.0         0.0
 ‚ãÆ                            ‚ã±                          
 0.0  0.0         0.0             0.368793    0.36933    -0.0120351
 0.0  0.0         0.0             0.789835    0.10943     0.0197234
 0.0  0.0         0.0         ‚Ä¶  -0.167317    0.206943    0.0121849
 0.0  0.0         0.0            -0.146398    0.193441   -0.0275316
 0.0  0.0         0.0            -0.0631539   0.0803796   0.0113878
 0.0  0.0         0.0            -0.0397994   0.0521778  -0.00356272
 0.0  0.0         0.0            -0.0309998   0.0109042  -0.00833901
 0.0  0.0         0.0         ‚Ä¶  -0.112318   -0.285862   -0.0291041
 0.0  0.0         0.0            -0.152088    0.362752   -0.0176148
 0.0  0.0         0.0             1.0         0.22565     0.0399238
 0.0  0.0         0.0             0.0         1.0        -0.0670657
 0.0  0.0         0.0             0.0         0.0         1.0</code></pre><p>Check if the given block is PSD by computing its eigenvalues</p><pre><code class="language-julia hljs">eigvals(Symmetric(Sigma)) # Symmetric uses upper triangular portion of data</code></pre><pre><code class="nohighlight hljs">10000-element Vector{Float64}:
  -2.5798836838860755
  -1.415006812481892
  -0.7674440501660297
  -0.6247787622218666
  -0.584928280772804
  -0.5435038109298787
  -0.5187701863140765
  -0.43166908059139875
  -0.41532154277852423
  -0.38813217585569226
  -0.3700017799653646
  -0.35667047980202393
  -0.3485121536359235
   ‚ãÆ
 109.56227091251577
 113.33023962897241
 127.38408283290424
 130.99148352578467
 132.82965433833886
 145.9473736360883
 155.27371456569082
 177.7213297373829
 194.1048422669041
 231.15057521884452
 293.44132970720796
 317.6410079580756</code></pre><p>Arbitrary slicing works but is very slow</p><pre><code class="language-julia hljs">@time bm[1:3, 1:2:100] # read columns 1, 3, 5, ...</code></pre><pre><code class="nohighlight hljs"> 30.278933 seconds (759 allocations: 13.219 KiB)





3√ó50 Matrix{Float64}:
 1.0  -0.0181891   0.00802797  -0.0421512  ‚Ä¶  0.0143534     0.0172358
 0.0  -0.00041394  0.00232164  -0.0112066     5.39059e-5   -0.0188801
 0.0   1.0         0.0         -0.0282011     0.000733417  -0.0277042</code></pre><h2 id="Read-in-a-block-with-get_block"><a class="docs-heading-anchor" href="#Read-in-a-block-with-get_block">Read in a block with <code>get_block</code></a><a id="Read-in-a-block-with-get_block-1"></a><a class="docs-heading-anchor-permalink" href="#Read-in-a-block-with-get_block" title="Permalink"></a></h2><p>One can also extract a block by specifying the chromosome and starting/ending basepair</p><pre><code class="language-julia hljs">chr = 1
start_pos = 57292
end_pos = 59193
sigma, df = get_block(bm, chr, start_pos, end_pos; min_maf=0.0)
sigma</code></pre><pre><code class="nohighlight hljs">8√ó8 Matrix{Float64}:
 1.0  0.0135236  -0.0249515  -0.0617588  ‚Ä¶  -0.00321314  -0.00718484
 0.0  1.0         0.0167518   0.0209895      0.00877351   0.012266
 0.0  0.0         1.0        -0.0125702     -0.0074448   -0.0373812
 0.0  0.0         0.0         1.0            0.0120626   -0.0486212
 0.0  0.0         0.0         0.0           -0.0186309   -0.0454317
 0.0  0.0         0.0         0.0        ‚Ä¶   0.900573     0.00806292
 0.0  0.0         0.0         0.0            1.0          0.00246611
 0.0  0.0         0.0         0.0            0.0          1.0</code></pre><pre><code class="language-julia hljs"># SNP information of this block
@show df;</code></pre><pre><code class="nohighlight hljs">df = 8√ó5 DataFrame
 Row ‚îÇ AF         chr      pos    ref     alt
     ‚îÇ Float64    String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ 0.0259568  1        57292  C       T
   2 ‚îÇ 0.992971   1        57952  A       C
   3 ‚îÇ 0.0586652  1        58176  G       A
   4 ‚îÇ 0.36321    1        58866  C       G
   5 ‚îÇ 0.0921872  1        59040  T       C
   6 ‚îÇ 0.010808   1        59108  G       A
   7 ‚îÇ 0.0112479  1        59121  G       T
   8 ‚îÇ 0.0370268  1        59193  T       G</code></pre><p>When importing blocks, one can filter for minimum minor allele frequency</p><pre><code class="language-julia hljs"># keep SNPs with MAF &gt; 0.05
chr = 1
start_pos = 57292
end_pos = 59193
sigma, df = get_block(bm, chr, start_pos, end_pos; min_maf=0.05)
sigma</code></pre><pre><code class="nohighlight hljs">3√ó3 Matrix{Float64}:
 1.0  -0.0125702  -0.0323548
 0.0   1.0        -0.0178985
 0.0   0.0         1.0</code></pre><pre><code class="language-julia hljs">@show df;</code></pre><pre><code class="nohighlight hljs">df = 3√ó5 DataFrame
 Row ‚îÇ AF         chr      pos    ref     alt
     ‚îÇ Float64    String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ 0.0586652  1        58176  G       A
   2 ‚îÇ 0.36321    1        58866  C       G
   3 ‚îÇ 0.0921872  1        59040  T       C</code></pre><p>One can also provide a list of SNP positions, and we will only keep SNPs that have those position which also pass the <code>min_maf</code> filter</p><pre><code class="language-julia hljs"># keep SNPs with MAF &gt; 0.05 and only include a list of SNPs with known positions
chr = 1
start_pos = 57292
end_pos = 59193
snps_to_keep = [58176, 58866]
sigma, df = get_block(bm, chr, start_pos, end_pos; min_maf=0.05, snps_to_keep=snps_to_keep)
sigma</code></pre><pre><code class="nohighlight hljs">2√ó2 Matrix{Float64}:
 1.0  -0.0125702
 0.0   1.0</code></pre><pre><code class="language-julia hljs">@show df;</code></pre><pre><code class="nohighlight hljs">df = 2√ó5 DataFrame
 Row ‚îÇ AF         chr      pos    ref     alt
     ‚îÇ Float64    String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ 0.0586652  1        58176  G       A
   2 ‚îÇ 0.36321    1        58866  C       G</code></pre><h2 id="SNP-information"><a class="docs-heading-anchor" href="#SNP-information">SNP information</a><a id="SNP-information-1"></a><a class="docs-heading-anchor-permalink" href="#SNP-information" title="Permalink"></a></h2><p>The <code>HailBlockMatrix</code> struct contains the SNP information for each row/column of the LD matrix. </p><pre><code class="language-julia hljs">@show bm.info[1:10, :]; # print first 10 columns</code></pre><pre><code class="nohighlight hljs">bm.info[1:10, :] = 10√ó5 DataFrame
 Row ‚îÇ AF          chr      pos    ref     alt
     ‚îÇ Float64     String3  Int64  String  String
‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îº‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
   1 ‚îÇ 0.409221    1        10146  AC      A
   2 ‚îÇ 0.00810811  1        10151  TA      T
   3 ‚îÇ 0.144444    1        10177  A       C
   4 ‚îÇ 0.0232558   1        10178  CCTAA   C
   5 ‚îÇ 0.00617284  1        10181  A       T
   6 ‚îÇ 0.0326087   1        10250  A       C
   7 ‚îÇ 0.0595238   1        10257  A       C
   8 ‚îÇ 0.198276    1        10327  T       C
   9 ‚îÇ 0.05        1        10329  AC      A
  10 ‚îÇ 0.00720461  1        10333  CT      C</code></pre><p>Alternatively, one can import this dataframe directly with the following syntax</p><pre><code class="language-julia hljs">df = read_variant_index_tables(ht_file)</code></pre></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../pan_UKB/">¬´ Pan UKBB</a><a class="docs-footer-nextpage" href="../api/">API ¬ª</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.5.0 on <span class="colophon-date" title="Monday 15 July 2024 02:42">Monday 15 July 2024</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
